InfrastructureBackupValidator/BackupValidationTool.psm1

```diff
@@ -0,0 +1,591 @@
+            $offers.Add($offerId, $offer)
+        }
+
+        Write-Verbose $Strings.MsgOffer -Verbose
+        foreach ($offer in $offers.Values)
+        {
+            Write-Host ($offer | Out-String)
+            Write-Host "`n"
+        }
+
+        # STEP 12: Retrieve subscriptions
+        $SQLCmd = "SELECT [SubscriptionState],[SubscriptionStateName] FROM [$SubscriptionSqlDbName].[subscriptions.internal].[SubscriptionStates]"
+        $subStateTable = Invoke-Sqlcmd -Database $SubscriptionSqlDbName -Query $SQLCmd -As DataSet @sqlCommonParams
+        $subscriptionStates = @{}
+        $subStateTable.Tables[0].Rows | % { $subscriptionStates.Add($_.SubscriptionState.ToString(), $_.SubscriptionStateName.ToString()) }
+
+        $SQLCmd = "SELECT [Id],[ResellerSubscriptionId],[SubscriptionId],[DisplayName],[OfferId],[Owner],[TenantId],[RoutingResourceManagerType],[State],[Tags] FROM [$SubscriptionSqlDbName].[subscriptions.internal].[Subscriptions]"
+        $subTable = Invoke-Sqlcmd -Database $SubscriptionSqlDbName -Query $SQLCmd -As DataSet @sqlCommonParams
+        $subscriptions = @{}
+        $subscriptionColumnNames = $subTable.Tables[0].Columns.ColumnName
+        foreach ($row in $subTable.Tables[0].Rows)
+        {
+            $subscriptionId = ""
+            $subscription = [ordered] @{}
+            foreach ($column in $subscriptionColumnNames)
+            {
+                $value = $row[$column].ToString()
+                if ($column -eq "ResellerSubscriptionId")
+                {
+                    $subscription[$column] = $resellerSubcriptions[$value]
+                }
+                elseif ($column -eq "State")
+                {
+                    $subscription[$column] = $subscriptionStates[$value]
+                }
+                elseif ($column -eq "RoutingResourceManagerType")
+                {
+                    $subscription[$column] = $resourceManagerTypes[$value]
+                }
+                elseif ($column -eq "OfferId")
+                {
+                    $subscription[$column] = $offers[$value].Id
+                }
+                elseif ($column -eq "Id")
+                {
+                    $subscriptionId = $value
+                }
+                else
+                {
+                    $subscription[$column] = $value
+                }
+            }
+
+            $subscription["Id"] = "/subscriptions/$($subscription["ResellerSubscriptionId"])/providers/Microsoft.Subscriptions.Admin/subscriptions/$($subscription["SubscriptionId"])"
+            $subscriptions.Add($subscriptionId, $subscription)
+        }
+
+        Write-Verbose $Strings.MsgSubscription -Verbose
+        foreach ($subscription in $subscriptions.Values)
+        {
+            Write-Host ($subscription | Out-String)
+            Write-Host "`n"
+        }
+
+        # STEP 13: Retrieve plans
```

Comment:
Nit: this is a gigantic function, maybe it would be easier to read if each step is in their own method.